src/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs

```diff
@@ -118,33 +118,42 @@ public void TFS_470021()
     </xs:complexType>
   </xs:element>
 </xs:schema>";
-            using (XmlWriter w = XmlWriter.Create("cham.xsd"))
+
+            using (var tempDirectory = new TempDirectory())
             {
-                using (XmlReader r = XmlReader.Create(new StringReader(cham)))
-                    w.WriteNode(r, true);
-            }
-            XmlSchemaSet ss = new XmlSchemaSet();
-            ss.ValidationEventHandler += new ValidationEventHandler(ValidationCallback);
+                string chamPath = Path.Combine(tempDirectory.Path, "cham.xsd");
+                string tempDirectoryPath = tempDirectory.Path[tempDirectory.Path.Length - 1] == Path.DirectorySeparatorChar ?
```

Comment:
This should not work on Windows. I've just tested this on Ubuntu and `ss.Add(null, XmlReader.Create(new StringReader(main), new XmlReaderSettings() { XmlResolver = new XmlUrlResolver() }, tempDirectoryPath));` is correctly resolving cham.xsd. Please file a bug and write a disabled test where you don't set it and you expect it to fail - I wouldn't fix it as part of this PR.